src/FPIndia/Jobs.hs

```diff
@@ -14,13 +14,85 @@ import System.UnionMount qualified as UM
 import UnliftIO
 
 data Job = Job
-  { jobName :: !Text
-  , jobLocation :: !Text
+  { jobName :: !String
+  , jobSource :: !SourceType
+  , jobWebsite :: !String
+  , --  , jobWebsite :: !SourceUrl
+    jobLocation :: !Text
+  , jobLanguages :: !Language
+  , jobPermalink :: !Text
+  , jobActiveStatus :: !Status
   }
-  deriving stock (Eq, Show)
+
+-- data Source = Source
+--   { sourceType :: SourceType
+--   , sourceUrl :: SourceUrl
+--   }
+
+data SourceType
+  = Reddit
+  | Github
+  | Twitter
+  | LinkedIn
+  | OtherJobSite
+  | OtherForum
+  | OtherSourceType
+
+--data SourceUrl = Verified String | Other String
+data Language
+  = Haskell
+  | PureScript
+  | Scala
+  | Ocaml
+  | Rust
+  | Erlang
+  | Agda
+  | Elixir
+  | Idris
```

Comment:
We need a clause for `| OtherLanguage String` where the string specifies the language name